File: connect6/Node.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace connect6
{
    public class Node
    {
        public State State { get; set; }
        public Node Parent { get; set; }
        public List<Node> ChildArray { get; set; }
        public List<Node> nodeList = new List<Node>();
       
        public Node()
        {

        }
        public Node(State state)
        {
            this.State = state;
            this.Parent = null;
            this.ChildArray = new List<Node>();
        }
        public State GetState()
        {
            return State;
        }
        public Node GetParent()
        {
            return this.Parent;
        }
        public void SetParent(Node parent)
        {
            this.Parent = parent;
        }

        public List<Node> GetChildArray()
        {

            return ChildArray;
        }

        public void SetChildArray(List<Node> childArray)
        {
            this.ChildArray = childArray;
        }
        public Node GetChildWithMaxScore()
        {
        
            if (ChildArray.Count == 0)
            {
                return null;
            }

            Node maxScoreNode = ChildArray[0];
            double maxScore = maxScoreNode.State.WinScore;

            foreach (var childNode in ChildArray)
            {
                if (childNode.State.WinScore > maxScore)
                {
                    maxScoreNode = childNode;
                    maxScore = childNode.State.WinScore;
                }
            }

            return maxScoreNode;
        }
        public Node GetRandomChildNode()
        {
            // Thực hiện logic để lấy một con ngẫu nhiên từ danh sách các con
            Random random = new Random();
            int randomIndex = random.Next(0, ChildArray.Count);
            return ChildArray[randomIndex];
        }


    }
}

File: src/Node.java
import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.ThreadLocalRandom;

public class Node {
	State state;
	Node parent;
	List<Node> childArray;

	public Node() {
		state = new State();
		childArray = new ArrayList<>();
	}

	public Node(State state) {
		this.state = state;
		childArray = new ArrayList<>();
	}

	public Node(Node node) {
		this.parent = node.getParent();
		this.childArray = node.getChildArray();
		this.state = new State(node.getState());
	}

	public State getState() {
		return state;
	}

	public void setState(State state) {
		this.state = state;
	}

	public Node getParent() {
		return parent;
	}

	public void setParent(Node parent) {
		this.parent = parent;
	}

	public List<Node> getChildArray() {
		return childArray;
	}

	public void setChildArray(List<Node> childArray) {
		this.childArray = childArray;
	}

	public Node getRandomChildNode() {
		return childArray.get(ThreadLocalRandom.current().nextInt(childArray.size()));
	}

	public Node getChildWithMaxScore() {
		if (childArray.size() == 0)
			return null;
		Node node = childArray.get(0);
		for (int i = 1; i < childArray.size(); i++)
			if (node.getState().getVisitCount() < childArray.get(i).getState().getVisitCount())
				node = childArray.get(i);
		return node;
	}

	@Override
	public String toString() {
		return "Node [state=" + state + "]";
	}
}
